### PR TITLE
Response for non-existing target

### LINKED ISSUE #702: Better response for non-existing/old target
Sentry Issue: [RED-HAT-0P-2MS](https://sentry.io/organizations/red-hat-0p/issues/1747750168/?referrer=github_integration)

```
CoprNoResultException: Build 1506070 was not submitted to fedora-29-x86_64 chroot.
(49 additional frame(s) were not displayed)
...
  File "copr/v3/helpers.py", line 59, in wrapper
    result = func(*args, **kwargs)
  File "copr/v3/helpers.py", line 59, in wrapper
    result = func(*args, **kwargs)
  File "copr/v3/proxies/build_chroot.py", line 22, in get
    response = request.send()
  File "copr/v3/requests.py", line 55, in send
    handle_errors(response)
  File "copr/v3/requests.py", line 106, in handle_errors
    raise CoprNoResultException(response_json["error"], response=response)
```

Triggered in this -- really old -- PR that  is rebuild without rebase for a long time:
- PR: https://github.com/packit-service/hello-world/pull/22
- config: https://github.com/packit-service/hello-world/blob/36c2ec2eb78e105da5fd99539c5a83c17954f70a/.packit.yaml

### PR DESCRIPTION
**Pull request by mfocko**
Fixes #702 

- [x] Set correct link to logs

### CODE REVIEW

File: packit_service/worker/build/copr_build.py [hunk 1]
```diff
@@ -214,6 +228,15 @@ def run_copr_build(self) -> TaskResults:
             )
 
         for chroot in self.build_targets:
+            if chroot not in self.available_chroots:
+                self.report_status_to_all_for_chroot(
+                    state=CommitStatus.error,
+                    description="Not supported target",
```
Comment (lachmanfrantisek) [inline packit_service/worker/build/copr_build.py:234]: Just to make it really clear? (The test name can be a bit cluttered and mixed with dashes.)

```suggestion
                    description=f"Not supported target: {chroot}",
```

File: packit_service/worker/build/copr_build.py [hunk 2]
```diff
@@ -173,6 +173,20 @@ def tests_targets(self) -> Set[str]:
...
+        Returns set of available COPR targets.
+        """
+        return {
+            *filter(
+                lambda chroot: not chroot.startswith("_"),
+                self.api.copr_helper.get_copr_client()
+                .mock_chroot_proxy.get_list()
+                .keys(),
+            )
+        }
```
Comment (TomasTomecek) [inline packit_service/worker/build/copr_build.py:188]: nice, this should make it trivial to implement the alias expansion

File: packit_service/worker/build/copr_build.py [hunk 3]
```diff
@@ -233,6 +258,18 @@ def run_copr_build(self) -> TaskResults:
...
             )
 
+        if unprocessed_chroots:
+            unprocessed = "\n".join(sorted(unprocessed_chroots))
+            available = "\n".join(sorted(self.available_chroots))
+            self.project.pr_comment(
+                pr_id=self.metadata.pr_id,
+                body="There are build targets that are not supported by COPR.\n"
+                "<details>\n<summary>Unprocessed build targets</summary>\n\n"
+                f"```\n{unprocessed}\n```\n</details>\n"
```
Comment (TomasTomecek) [inline packit_service/worker/build/copr_build.py:268]: this should be fairly short most of the time right? would it make sense to print it right away without the `<details>` thing?

**Review by TomasTomecek — APPROVED** [review]
LGTM, one question

**Review by sakalosj — APPROVED** [review]

File: packit_service/worker/build/copr_build.py (hunk 3 above)
Comment (mfocko) [reply packit_service/worker/build/copr_build.py:268]: I'm not really sure...
- for unprocessed: I could smack there easily a lot of invalid targets, or if I wanted to be explicit just use _vim-magic_ and have a typo somewhere in there; that would also generate long list
- for supported: I just looked at the list from the code snippet, there's 42 of them

**Pull request merged**